src/scan-options.ts

```diff
@@ -55,8 +61,8 @@ export type KeyTypeForScanOptions<O extends ScanOptions> = O extends {
   : string;
 
 /**
- * When providing a start key for an indexed [[scan]] you need to provide the
- * `secondary` index to start [[scan]] at. You may also provide a `primary`
+ * When providing a start key for an indexed [[ReadTransaction.scan|scan]] you need to provide the
+ * `secondary` index to start [[ReadTransaction.scan|scan]] at. You may also provide a `primary`
```

Comment:
The key to start scanning at.

If you are scanning the primary index (i.e., you did not specify [[indexName]]), then pass a single string for this field, which is the key in the primary index to scan at.

If you are scanning a secondary index (i.e., you specified [[indexName]]), then use the tuple form. In that case, `secondary` is the secondary key to start scanning at, and `primary` (if any) is the primary key to start scanning at.

Reply:
This is pretty intricate. Is it possible to modify the types to make it so that only the non-tuple form is allowed if `indexName` is specified?

Reply:
> This is pretty intricate. Is it possible to modify the types to make it so that only the non-tuple form is allowed if indexName is specified?

Created a new issue. It seems doable.